backend/app/training/scrape_products.py

```
@@ -251,18 +252,54 @@ async def scrape_products(sitemap_url, scrape_limit=float('inf'), sample_size=5,
+            print(f"Error scraping {sitemap}: {e}")
+            return []
+
+    # 1) Try auto-discovery
+    discovered = find_sitemaps(website_url)
+    if discovered:
+        first = discovered[0]
+        products = await _attempt(first)
+        if products:
+            print(f"Success with discovered sitemap: {first}")
+            return products
+        else:
```

Ruff (0.8.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)